utils/log/progress.go

```diff
@@ -0,0 +1,11 @@
+package log
+
+import "io"
+
+type ProgressBar interface {
+	NewBar(total int64, prefix, description string) (barId int)
+	NewBarReplacement(replaceBarId int, prefix, description string) (barId int)
```

Comment:
replaceBarId --> replaceId
NewBarReplacemen --> NewReplacement
barId --> id

The above method and vars probably need a description.